api/libproject/devstart.rb

```diff
@@ -1136,3 +1116,72 @@ def docker_run(cmd_name, args)
   :description => "Runs the specified command in a docker container.",
   :fn => lambda { |*args| docker_run("docker-run", args) }
 })
+
+def create_project_resources(gcc)
+  common = Common.new
+  common.status "Enabling APIs..."
+  for service in SERVICES
+    common.run_inline("gcloud service-management enable #{service} --project #{gcc.project}")
+  end
+  common.status "Creating GCS bucket to store credentials..."
+  common.run_inline("gsutil mb -p #{gcc.project} -c regional -l us-central1 gs://#{gcc.project}-credentials/")
+  common.status "Creating Cloud SQL instances..."
+  common.run_inline("gcloud sql instances create #{INSTANCE_NAME} --tier=db-n1-standard-2 " +
+                    "--activation-policy=ALWAYS --backup-start-time 00:00 " +
+                    "--failover-replica-name #{FAILOVER_INSTANCE_NAME} --enable-bin-log " +
+                    "--database-version MYSQL_5_7 --project #{gcc.project} --storage-auto-increase")
+  common.status "Creating AppEngine app..."
+  common.run_inline("gcloud app create --region us-central --project #{gcc.project}")
+end
+
+def setup_project_data(gcc, root_password, workbench_password, public_password, args)
+  common = Common.new
+  # This changes database connection information; don't call this while the server is running!
+  common.status "Writing DB credentials file..."
+  write_db_creds_file(gcc.project, root_password, workbench_password, public_password)
+  common.status "Setting root password..."
+  run_with_redirects("gcloud sql users set-password root % --project #{gcc.project} " +
+                     "--instance #{INSTANCE_NAME} --password #{root_password}",
+                     to_redact=root_password)
+  # Don't delete the credentials created here; they will be stored in GCS and reused during
+  # deployment, etc.
+  do_run_with_creds(gcc.project, gcc.account, nil, false) do |creds_file|
+    with_cloud_proxy_and_db_env("setup-db-users", args) do |ctx|
+      common.status "Copying service account key to GCS..."
+      common.run_inline("gsutil cp #{creds_file} gs://#{gcc.project}-credentials/app-engine-default-sa.json")
```

Comment:
I left a comment in the doc which wasn't answered. Why would it be necessary to create/store this key in GCS? There should be a very compelling reason since this will likely be our most privileged SA.

Reply:
We need to store a service account key in GCS for use when talking to directory service. I'm not sure it necessarily *has* to be the AppEngine service account though... @dmohs could we make it another service account with basically no permissions?

For now, making it be the AE service account... we can hopefully change this later.

Reply:
There shouldn't be any such restriction, at least on the gsuite side. We should definitely switch this to be a different SA. From a thread I found from @jmandel explaining the directory service interaction, he described this as a different SA from the GAE default account.

Sure, feel free to punt unless you think this will cause more problems down the road.

Reply:
Correct. We currently DO use our default service account, but I know of no reason we HAVE to use it. I am strongly in favor of creating a new account that is scoped for this one thing.